AsyncDisplayKit/Details/ASCollectionDataController.mm

```diff
@@ -143,15 +145,14 @@ - (void)prepareForInsertRowsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths
 
 - (void)willInsertRowsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths
 {
-  NSArray *keys = _pendingContexts.allKeys;
-  for (NSString *kind in keys) {
+  for (NSString *kind in _pendingContexts) {
     NSMutableArray<ASIndexedNodeContext *> *contexts = _pendingContexts[kind];
```

Comment:
Nice find! Through testing, I found that if you subclass NSDictionary then enumerateKeysAndObjects: will call __NSDictionaryEnumerate which calls objectForKey: for each key (as it is required to!). But if you don't subclass, the enumeration method uses its own fast path.